### ISSUE TITLE
Unable to set accessibility name on SettingsCard's HeaderIcon image

### ISSUE THREAD

**Issue by guimafelipe** [CommunityToolkit/Windows#552]
### Describe the bug

The SettingsCard control have an accessibility issue that makes it unable to set an accessible name for the image used as HeaderIcon by the control, even when it is set manually when instantiating it.

### Steps to reproduce

Create a SettingsCard control with an image as a HeaderIcon (there is one in the sample gallery).
Add a AutomationProperties.Name to the image in the XAML.

### Expected behavior

The image will have its own accessible name as it was set up in the XAML, that can be narrated when keyboard focused. But there is none.

### Code Platform

- [ ] UWP
- [x] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [x] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [x] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [ ] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Visual Studio Version

2022

### Device form factor

Desktop

### Help us help you

No, I'm unable to contribute a solution.
